### DISCUSSION TITLE
How to add an internal module

### DISCUSSION THREAD

**Discussion by Melab** [nodejs/help#4900] [Q&A] [answered]
To add a built-in module to the code, what code must be modified?

**Comment by alex-semenyuk** [comment]
Suppose this guide will be helpful https://www.digitalocean.com/community/tutorials/how-to-create-a-node-js-modulev

**Reply by olivermrose** [reply]
They're asking how to create a native module like `fs` and `path`, not how to create an npm package.

**Reply by Melab** [reply]
> Suppose this guide will be helpful https://www.digitalocean.com/community/tutorials/how-to-create-a-node-js-modulev

Not what I mean. Please reread my question.

**Comment by aduh95** [comment, marked as answer]
Clone nodejs/node, add a `.js` file in `lib/` folder, and run `make -j12` (depends on your OS/platform, see https://github.com/nodejs/node/blob/HEAD/BUILDING.md for further instructions): voilà. You would now have a node executable at `out/Release/node` with the new built-in module you added.

If you're asking how that module could land on Node.js land, you would need to open a PR on this repo, adding docs, tests, and explanation why you think this module should be part of core. See https://github.com/nodejs/node/blob/HEAD/doc/guides/modules-in-core.md for more info.
